server/src/main/java/org/elasticsearch/gateway/MetaStateService.java

```diff
@@ -85,7 +85,7 @@ public IndexMetaData loadIndexState(Index index) throws IOException {
      */
     List<IndexMetaData> loadIndicesStates(Predicate<String> excludeIndexPathIdsPredicate) throws IOException {
         List<IndexMetaData> indexMetaDataList = new ArrayList<>();
-        for (String indexFolderName : nodeEnv.availableIndexFolders()) {
+        for (String indexFolderName : nodeEnv.availableIndexFolders(excludeIndexPathIdsPredicate)) {
             if (excludeIndexPathIdsPredicate.test(indexFolderName)) {
                 continue;
```

Comment:
I think something has gone badly wrong if we end up here. Can we at least `assert false`, and maybe throw something in prod code too?